models/models.js

```diff
@@ -54,35 +54,54 @@ function selectCommentsByArticleId(articleId) {
     });
 }
 
-function checkArticleIdExists(articleId) {
-  return db
-    .query(`SELECT title FROM articles WHERE article_id = $1`, [articleId])
-    .then(({ rows }) => {
-      if (rows.length === 0) {
-        return Promise.reject({ status: 404, msg: "not found" });
-      }
-    });
-}
-
 function insertCommentByArticleId(articleId, newComment) {
   return checkIfUserExists(newComment.username)
     .then(() => {
-      return db
-        .query(
-          ` INSERT INTO comments (body, author, article_id)
+      return db.query(
+        ` INSERT INTO comments (body, author, article_id)
             VALUES ($1, $2, $3)
             RETURNING * ;`,
-          [newComment.body, newComment.username, articleId]
-        )
-        .then(({ rows }) => {
-          return rows[0];
-        });
+        [newComment.body, newComment.username, articleId]
+      );
+    })
+    .then(({ rows }) => {
+      return rows[0];
     })
     .catch((err) => {
       return Promise.reject(err);
     });
 }
 
+function updateArticleById(articleId, newVote) {
+  return checkArticleIdExists(articleId)
```

Comment:
This would be a great use of a promise.all rather than a longer chain. If checkArticle rejects, the whole thing rejects.